### PR TITLE
chore(RHTAPWATCH-662): add inputs to odcs compose entrypoint

### PR DESCRIPTION
**Pull request by yftacherzog**
Required inputs when running the odcs entrypoint to specify where the compose should be stored and where the configurations reside.

Had to rename the directory name to avoid it having the same name as one of the the modules.

### CODE REVIEW

**Comment by srivickynesh** [comment]
@yftacherzog could you check with removing the full python version from Pipfile. sometimes we might have different version in 3.11. but it requires to be in 3.11.6.

**Comment by yftacherzog** [comment]
> @yftacherzog could you check with removing the full python version from Pipfile. sometimes we might have different version in 3.11. but it requires to be in 3.11.6.

I added the requirement on workload as you added it on the pipfile. Since then I didn't see failures. Was there a specific reason to have 3.11.6 on the pipfile? Do you suggest we make 3.11 on both the workflow and the pipfile?

**Comment by srivickynesh** [comment]
full python version is generated automatically when I created the Pipfile. yes, we can change it in both places.

**Comment by yftacherzog** [comment]
> full python version is generated automatically when I created the Pipfile. yes, we can change it in both places.

I changed it. For now it didn't fail. I suspect what happened before is that sometimes the workflow would create a 3.11.6 env by chance and then it would work, but sometimes it created another 3.11 version and then pipenv failed to find the 3.11.6 version.

:crossed_fingers:

**Comment by yftacherzog** [comment]
@Omeramsc please take a look

File: generate_compose/odcs_compose_generator.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    container_data: dict = yaml.safe_load(container_yaml_path.read_text())
+    content_sets_data: dict = yaml.safe_load(content_sets_yaml_path.read_text())
+
+    compose_generator = ComposeGenerator(
+        configurations_generator=ODCSConfigurationsGenerator(
+            container_data=container_data,
+            content_sets_data=content_sets_data,
+        ),
+        requestor=ODCSRequester(),
+        fetcher=ODCSFetcher(compose_path=compose_path),
```
Comment (Omeramsc) [inline generate_compose/odcs_compose_generator.py:53]: If you want it to fit my change, I'll be renaming it to `compose_file_path`

File: generate_compose/odcs_compose_generator.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from .compose_generator import ComposeGenerator
+from .odcs_configurations_generator import ODCSConfigurationsGenerator
+from .odcs_fetcher import ODCSFetcher
+from .odcs_requester import ODCSRequester
+
+
+@click.command()
+@click.option(
+    "--compose-path",
+    help="Path to where the compose should be saved.",
```
Comment (Omeramsc) [inline generate_compose/odcs_compose_generator.py:19]: Gonna be the path to the compose file itself

File: tests/test_odcs_compose_generator.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        self,
+        compose_path: Path,
+        container_data: dict,
+        container_yaml: Path,
+        content_sets_data: dict,
+        content_sets_yaml: Path,
+        mock_compose_generator: MagicMock,
+        mock_config_generator: MagicMock,
+        mock_requester: MagicMock,
+        mock_fetcher: MagicMock,
```
Comment (Omeramsc) [inline tests/test_odcs_compose_generator.py:110]: I wonder if that heavy usage of monkeypatching may be hinting us at a future dependency injection refactor.

**Review by Omeramsc — APPROVED** [review]
LGTM, left some minor comments.

File: tests/test_odcs_compose_generator.py (hunk 3 above)
Comment (yftacherzog) [reply tests/test_odcs_compose_generator.py:110]: We are using dependency injection. Those tests for the entrypoint script is the only place where we'd have to use monkeypatch, but I can see if I can make those as inputs to the script as well.

File: generate_compose/odcs_compose_generator.py (hunk 1 above)
Comment (yftacherzog) [reply generate_compose/odcs_compose_generator.py]: done

File: generate_compose/odcs_compose_generator.py (hunk 2 above)
Comment (yftacherzog) [reply generate_compose/odcs_compose_generator.py]: done

File: tests/test_odcs_compose_generator.py (hunk 3 above)
Comment (yftacherzog) [reply tests/test_odcs_compose_generator.py:110]: I ran into some trouble passing the injected dependency to the script main function during tests, which also needs to take click command line arguments.
We did some somewhat similar things in the past, but for some reason the unit test invoked the function with just the command line arguments and not the injected dependencies.

As the current pattern of using monkeypatch only on the command line script is what we did in the past as well, I'm going to leave it as-is for now.

**Pull request merged**